snapcraft/remote/git.py

```diff
@@ -50,6 +61,41 @@ def is_repo(path: Path) -> bool:
         ) from error
 
 
+def get_git_repo_type(path: Path) -> GitType:
+    """Check if a directory is a git repo and return the type.
+
+    :param path: filepath to check
+
+    :returns: GitType
+    """
+    if is_repo(path):
+        repo = pygit2.Repository(path)
+        if repo.is_shallow:
+            return GitType.SHALLOW
+        return GitType.NORMAL
+
+    return GitType.INVALID
+
+
+def check_git_repo_for_remote_build(path: Path) -> None:
+    """Check if a directory meets the requirements of doing remote builds.
+
+    :param path: filepath to check
+
+    :raises SnapcraftError: if incompatible git repo is found
+    """
+    git_type = get_git_repo_type(path.absolute())
+
+    if git_type == GitType.INVALID:
+        raise SnapcraftError(f"Could not find a git repository in {str(path)!r}")
+
+    if git_type == GitType.SHALLOW:
+        raise SnapcraftError(
```

Comment:
This should be a `RemoteBuild` error and not mention anything specific to snapcraft or its bases (since this module will get migrated out of snapcraft into a library and used by other applications)